### PR TITLE
Decorate all domain collection container for emitting build ops

### PR DESCRIPTION
**Pull request by breskeby**
### Context

- Follow up on https://github.com/gradle/gradle/pull/7734 on decorating *all* domain collection containers for emitting build operations when callback action is executed. 

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status

### CODE REVIEW

**Comment by breskeby** [comment]
I added a number of more integration tests to verify those changes but I don't cover _all_ containers as this would be too much overhead IMO and this is mostly a mechanic change in the wiring of the containers as the general infrastructure this is based on was already done in an earlier PR (https://github.com/gradle/gradle/pull/7734)

I've seen, even after we put the decoration behind a feature toggle, this change came with some performance cost. I don't see we can do much about it as part of this change IMO

**Comment by breskeby** [comment]
build is passing. I needed to bring back some constructors of internal classes as they're used by smoke tested plugins.

File: subprojects/core/src/test/groovy/org/gradle/api/internal/DefaultPolymorphicDomainObjectContainerTest.groovy [hunk 1]
```diff
@@ -30,7 +30,7 @@ class DefaultPolymorphicDomainObjectContainerTest extends AbstractPolymorphicDom
     def agedFred = new DefaultAgeAwarePerson(name: "fred", age: 42)
     def agedBarney = new DefaultAgeAwarePerson(name: "barney", age: 42)
 
-    def container = new DefaultPolymorphicDomainObjectContainer<Person>(Person, DirectInstantiator.INSTANCE)
+    def container = new DefaultPolymorphicDomainObjectContainer<Person>(Person, DirectInstantiator.INSTANCE, CollectionCallbackActionDecorator.NOOP)
 
     final boolean supportsBuildOperations = false
```
Comment (ldaley) [inline subprojects/core/src/test/groovy/org/gradle/api/internal/DefaultPolymorphicDomainObjectContainerTest.groovy]: This needs to go to true, and the container should use the decorator from the base class.

This was the only test not supporting operations before, so this flag needs to be removed now and all containers must support it.

Comment (breskeby) [reply subprojects/core/src/test/groovy/org/gradle/api/internal/DefaultPolymorphicDomainObjectContainerTest.groovy]: fixed

**Comment by breskeby** [comment]
Merged into release after syncing with @oehme. 
Feedback from @ldaley was addressed

**Pull request merged**